modules/rules_cue/0.2.0/MODULE.bazel

```diff
@@ -0,0 +1,31 @@
+module(
+    name = "rules_cue",
+    version = "0.2.0",
+)
+
+bazel_dep(name = "bazel_skylib", version = "1.3.0")
+bazel_dep(name = "platforms", version = "0.0.6")
+bazel_dep(name = "rules_go", version = "0.38.1")
+
+cue = use_extension("//cue:extensions.bzl", "cue")
+use_repo(
+    cue,
+    "cue_tool_toolchains",
+)
+
+register_toolchains("@cue_tool_toolchains//:all")
+
+# Tailor the Go SDK that we use here.
+go_sdk = use_extension("@rules_go//go:extensions.bzl", "go_sdk")
+go_sdk.download(
```

Comment:
Could you briefly explain why this needs to bring in a custom Go toolchain?

Reply:
I suppose that it doesn't _need_ to do so. My inclination was to make sure that it works well with the newest toolchain available, to reduce the impetus to update it soon.

I'm willing to remove this override, though, if you think that it's uncalled for, or that it perturbs callers' repositories. You will know better than I do at this point what the consequences are for these callers.

Reply:
You can mark the `use_extension` call as `dev_dependency = True` to achieve that.

Picking up toolchains from transitive dependencies can be surprising, which is we are currently how this works for rules_go. Better if it isn't needed here :⁠-⁠)

Reply:
Or maybe that fails in register_toolchain, but we should fix that then.

Reply:
As you predicted, using `dev_dependency = True` in the `use_extension` call causes this subsequent failure:
```
ERROR: /<some path>/examples/bzlmod/root/BUILD.bazel:6:30:
  While resolving toolchains for target //root:root:
  invalid registered toolchain '@go_custom_sdk_toolchains//:all':
  No repository visible as '@go_custom_sdk_toolchains' from repository '@rules_cue~override'
```

Reply:
In the next version of rules_go, you will be able to register toolchains without `use_repo` and `register_toolchain`, which works around this limitation.

@Wyverald What do you think, should `register_toolchains` and `register_execution_platforms` also come with a `dev_dependency` keyword argument?

Reply:
> What do you think, should register_toolchains and register_execution_platforms also come with a dev_dependency keyword argument?

What's the use case? It really feels like we're slapping `dev_dependency` on everything and I wonder if it's bad to reverse course and just add `if` statements...

Reply:
Regarding the `if` vs. `dev_dependency`: While the latter can appear repetitive, I think it's very consistent, boring in the right way and tool friendly. Having `if` available could easily end up letting people write WORKSPACE-style spaghetti.